docs/dev-guide.md

```diff
@@ -0,0 +1,292 @@
+$ export CAPSULE_IMAGE_VESION=`docker images --format '{{.Tag}}' quay.io/clastix/capsule`
+
+# If k3s, load the image into cluster by
+$ k3d image import --cluster k3s-capsule capsule quay.io/clastix/capsule:${CAPSULE_IMAGE_VESION}
+# If Kind, load the image into cluster by
+$ kind load docker-image --name kind-capsule quay.io/clastix/capsule:${CAPSULE_IMAGE_VESION}
+
+# deploy all the required manifests
+# Note: 1) please retry if you saw errors; 2) if you want to clean it up first, run: make remove
+$ make deploy
+
+# Make sure the controller is running
+$ kubectl get pod -n capsule-system
+NAME                                          READY   STATUS    RESTARTS   AGE
+capsule-controller-manager-5c6b8445cf-566dc   1/1     Running   0          23s
+
+# Check the logs if needed
+$ kubectl -n capsule-system logs --all-containers -l control-plane=controller-manager
+
+# You may have a try to deploy a Tenant too to make sure it works end to end
+$ kubectl apply -f - <<EOF
+apiVersion: capsule.clastix.io/v1beta1
+kind: Tenant
+metadata:
+  name: oil
+spec:
+  owners:
+  - name: alice
+    kind: User
+EOF
+
+# There shouldn't be any errors and you should see the newly created tenant
+$ kubectl get tenants
+NAME   STATE    NAMESPACE QUOTA   NAMESPACE COUNT   NODE SELECTOR   AGE
+oil    Active                     0                                 14s
+```
+
+## Scaling down the deployed Pod
+
+As of now, a complete env has been set up in `kind` or `k3d` powred cluster, and the `capsule-controller-manager` is running as a deployment serving as:
+
+- The reconcilers for CRDs and;
+- A series of webhooks 
+
+During development, we prefer the code is running within our IDE locally, so the `capsule-controller-manager` running in the Kubernetes will competing with us.
+
+To avoid that, we need to scale the existing replicas of `capsule-controller-manager` to 0:
+
+```sh
+$ kubectl -n capsule-system scale deployment capsule-controller-manager --replicas=0
+deployment.apps/capsule-controller-manager scaled
+```
+
+## Preparing TLS certificate for webhooks
+
+Running webhooks requires TLS, so let's prepare the TLS key pair in our development env to handle HTTPS requests.
+
+```sh
+# Create this dir to mimic the Pod mount point
+mkdir -p /tmp/k8s-webhook-server/serving-certs
+
+# Generate cert/key under /tmp/k8s-webhook-server/
+openssl req -newkey rsa:4096 -days 3650 -nodes -x509 \
+  -subj "/C=SG/ST=SG/L=SG/O=CAPSULE/CN=CAPSULE" \
+  -extensions SAN \
+  -config <( cat $( [[ "Darwin" -eq "$(uname -s)" ]]  && echo /System/Library/OpenSSL/openssl.cnf || echo /etc/ssl/openssl.cnf  ) \
+    <(printf "[ SAN ]\nsubjectAltName = \"IP:${LAPTOP_HOST_IP}\"")) \
+  -keyout /tmp/k8s-webhook-server/serving-certs/tls.key \
+  -out /tmp/k8s-webhook-server/serving-certs/tls.crt
+```
+
+Do note that there is an [known issue in controller-runtime, #900](https://github.com/kubernetes-sigs/controller-runtime/issues/900) when you now try to run the code locally:
```

Comment:
Let's put more emphasis on this issue since it's expected to happen only for non-Linux users.

Reply:
I guess you meant **put less emphasis**, instead of **more** on the issue. If so, yep, will remove that part.

Reply:
My bad, I didn't explain myself correctly 😅 

I was referring to something like [this](https://kind.sigs.k8s.io/docs/user/known-issues/): we're aware of some issues, we can easily document them so people don't waste their time looking for a solution.

Maybe a separate file, or a separate MD section?

Reply:
I was thinking that doc may potentially attract more Google search hits too :)

Having a **Known Issues** section makes a lot of sense when Capsule has to work around them.
But this is an upstream issue so we may simply just have the direct solution documented -- simplicity is in the heart of Capsule, you know better than me on this :)